src/Components/DashBoard/Repository/RepoComponents/RepoDetails/RepoDetailBackdrop/RemoteConfigComponent/RemoteManagementComponent.js

```diff
@@ -16,19 +15,38 @@ export default function AddRemoteRepoComponent({ repoId }) {
   const [deleteFailed, setDeleteFailed] = useState(false);
   const [addNewRemote, setAddNewRemote] = useState(true);
   const [addRemoteStatus, setAddRemoteStatus] = useState(false);
+  const [statusCheck, setStatusCheck] = useState(false);
+  const [remoteOperation, setRemoteOperation] = useState("add");
   const [remoteDetails, setRemoteDetails] = useState([]);
 
   useEffect(() => {
     let remoteArray = [];
-    //TODO: Add API call here and load it into the state
-
-    for (let i = 0; i < localStorage.length; i++) {
-      var key = localStorage.key(i);
-      var value = JSON.parse(localStorage.getItem(key));
-      remoteArray.push(value);
-    }
-
-    setRemoteDetails([...remoteArray]);
+    axios({
+      url: globalAPIEndpoint,
+      method: "POST",
+      data: {
+        query: `
+              query {
+                getRemote(repoId: "${repoId}"){
+                  remoteDetails
+                }
+              }
+            `,
+      },
+    })
+      .then((res) => {
+        res.data.data.forEach((items) => {
```

Comment:
`res.data.data` will hold the `remoteDetails` array. So `forEach` has to be used for that 

``` javascript
res.data.data.remoteDetails.forEach((items) => {...})
```